src/main/java/Duke.java

```diff
@@ -1,10 +1,46 @@
+/**
+ * The Duke class
+ */
 public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Creates a Duke object.
+     *
+     * @param filePath A string representing the directory of the file
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        tasks = new TaskList(storage.load());
+    }
+
+    /**
+     * Loads and saves files by applying other classes.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
```

Comment:
Systematic coding of each individual action that you want to execute to a corresponding method.